src/main/java/com/impactupgrade/nucleus/util/PageResultsProcessor.java

```diff
@@ -0,0 +1,31 @@
+package com.impactupgrade.nucleus.util;
+
+import com.google.common.base.Strings;
+import com.impactupgrade.nucleus.model.PagedResults;
+
+import java.util.function.Consumer;
+
+public class PageResultsProcessor<T> {
+
+  private final Consumer<PagedResults.ResultSet<T>> consumer;
+  private final CheckedFunction<String, PagedResults.ResultSet<T>> getter;
+
+  public PageResultsProcessor(Consumer<PagedResults.ResultSet<T>> consumer,
+                              CheckedFunction<String, PagedResults.ResultSet<T>> getter) {
+    this.consumer = consumer;
+    this.getter = getter;
+  }
+
+  public void process(PagedResults.ResultSet<T> resultSet) throws Exception {
```

Comment:
@VSydor Could we have this be `process(PagedResults<t>)` instead, and then this processor could do the ResultSet for loop rather than having each caller do that as well?

Reply:
Sure - applying the change.